riscv/csrs.cc

```diff
@@ -741,8 +742,8 @@ bool misa_csr_t::unlogged_write(const reg_t val) noexcept {
   proc->set_extension_enable(EXT_ZCMT, proc->extension_enabled(EXT_ZCA));
   proc->set_extension_enable(EXT_ZFH, new_misa & (1L << ('F' - 'A')));
   proc->set_extension_enable(EXT_ZFHMIN, new_misa & (1L << ('F' - 'A')));
-  proc->set_extension_enable(EXT_ZVFH, (new_misa & (1L << ('V' - 'A'))) && proc->extension_enabled(EXT_ZFHMIN));
-  proc->set_extension_enable(EXT_ZVFHMIN, new_misa & (1L << ('V' - 'A')));
+  proc->set_extension_enable(EXT_ZVFH, new_v && proc->get_isa().get_zvf() && proc->extension_enabled(EXT_ZFHMIN));
+  proc->set_extension_enable(EXT_ZVFHMIN, new_v && proc->get_isa().get_zvf());
```

Comment:
```suggestion
proc->set_extension_enable(EXT_ZVFHMIN, new_v && proc->get_isa().get_zvf());
proc->set_extension_enable(EXT_ZVFH, new_v && proc->get_isa().get_zvf() && proc->extension_enabled(EXT_ZFHMIN));
```
Does the order matter here? I think that if EXT_ZVFHMIN is currently disabled but you plan to enable it, trying to enable Zvfh first might fail, because Zvfh depends on Zvfhmin being enabled.

Reply:
Yes, you are correct  and the order matters.  I will fix it

Reply:
@nadime15 correct my previous statement.

`EXT_ZVFH` depend on `EXT_ZFHMIN` but not `EXT_ZVFHMIN`

Reply:
Ups, you are right, I read Ext_Zfhmin as Ext_Zvfhmin!

Reply:
This is unrelated to the PR itself, but I noticed this line:

  `proc->set_extension_enable(EXT_ZCF, (new_misa & (1L << ('F' - 'A'))) && proc->extension_enabled(EXT_ZCA));`

Since Zcf is RV32 only, shouldn this check xlen as well (unless its checked somewhere else) ? @aswaterman

Reply:
> Since Zcf is RV32 only, shouldn this check xlen as well (unless its checked somewhere else) ?

I agree it should be changed to check XLEN == 32.  Please submit a separate PR for that one.

It isn't actually a bug right now, because, while this would cause `register_base_instructions` to do the wrong thing, `register_base_instructions` is only called during the construction of `processor_t`, before `misa` can be changed.  But if we ever add support for dynamically changing XLEN, then we'd need to reinvoke `register_base_instructions`, at which point this would become a problem.